### ISSUE TITLE
"imgproc.cpp issue on MacOS #30" is happen to me again, and i find answer

### ISSUE THREAD

**Issue by iamGreedy** [hybridgroup/gocv#57]
I find this line on ./env.sh
```
CVPATH=$(brew info opencv | sed -n "4p" | sed -e "s/ (.*//g")
```

But i already install opencv 2 and 3.1
So i have 3 opencv version 2, 3.1, 3.4

It cause  set CVPATH="/usr/local/Cellar/opencv/2.4.13"

So i changed 

```
CVPATH=$(brew info opencv | grep -E "opencv/3\.([3-9]|[1-9]\d{1,})" | sed -e "s/ (.*//g")
```
This make CVPATH can select cv3.3 above
And it seem fine

If anyone have multiple opencv installed by brew can have same problem

**Comment by deadprogram** [comment]
Hello @iamGreedy thanks for the info. Perhaps you would like to change the env.sh file and submit a PR with this change?

**Comment by tectiv3** [comment]
```
$ ls /usr/local/Cellar/opencv/3.3.
3.3.0_3/ 3.3.1_1/
```
This bug strikes again! I have two v3 opencv's :))

**Comment by deadprogram** [comment]
The regex is not specific enough, it seems. In this case 2 versions of OpenCV that match 3.3.x are installed. Perhaps we should tell people to upgrade to a single version, or else modify the regex further.

**Comment by tectiv3** [comment]
yeah, I think if I do `brew purge` or something it will delete older one. But I think fixing regex is a better way. I will try to fix it myself.

**Comment by iamGreedy** [comment]
How about using go instead shell?
I write simple code 
```go
package main
import (
	"runtime"
	"fmt"
	"os"
	"regexp"
	"strings"
	"os/exec"
	"io/ioutil"
)
var (
	macosRegexp = regexp.MustCompile(`/usr/local/Cellar/opencv/3\..* `)
)
func main(){
	
	switch runtime.GOOS{
	case "windows":
		fmt.Println("No support for Windows")
	case "darwin":	
		CVPATH, err := macOS()
		if err != nil{
			fmt.Errorf("%e", err)
			return
		}
		os.Setenv("CGO_CPPFLAGS", strings.Replace("-I$CVPATH/include -I$CVPATH/include/opencv2", "$CVPATH", CVPATH, -1))
		os.Setenv("CGO_CXXFLAGS", strings.Replace("--std=c++1z -stdlib=libc+", "$CVPATH", CVPATH, -1))
		os.Setenv("CGO_LDFLAGS", strings.Replace(
			"-L$CVPATH/lib -lopencv_core -lopencv_videoio -lopencv_imgproc -lopencv_highgui -lopencv_imgcodecs -lopencv_objdetect -lopencv_features2d -lopencv_video -lopencv_xfeatures2d", 
			"$CVPATH", CVPATH, -1))
		fmt.Println("Environment variables configured for macOS")
	case "linux":
		os.Setenv("CGO_CXXFLAGS", "--std=c++1z")
		if _, err := os.Stat("/etc/pacman.conf"); os.IsExist(err) {
			os.Setenv("CGO_CPPFLAGS", "-I/usr/include")
			os.Setenv("CGO_LDFLAGS", "-L/lib64 -lopencv_core -lopencv_face -lopencv_videoio -lopencv_imgproc -lopencv_highgui -lopencv_imgcodecs -lopencv_objdetect -lopencv_features2d -lopencv_video -lopencv_xfeatures2d")
		}else{
			os.Setenv("CGO_CPPFLAGS", "-I/usr/local/include")
			os.Setenv("CGO_LDFLAGS", "-L/lib64 -lopencv_core -lopencv_face -lopencv_videoio -lopencv_imgproc -lopencv_highgui -lopencv_imgcodecs -lopencv_objdetect -lopencv_features2d -lopencv_video -lopencv_xfeatures2d")
		}
		fmt.Println("Environment variables configured for Linux")
	case "freebsd":
		fallthrough
	default:
		fmt.Println("Unknown platform '" + runtime.GOOS + "'!")
	}
}

func macOS() (path string ,err error){
	cmd := exec.Command("brew", "info", "opencv")
	if err = cmd.Run(); err != nil{
		return 
	}
	bts, err := ioutil.ReadAll(cmd.Stdin)
	if err != nil{
		return 
	}
	var version Version
	for _, v  := range macosRegexp.FindAll(bts, -1){
		var minor, patch, ex int
		fmt.Sscanf(string(v), "/usr/local/Cellar/opencv/3.%d.%d_%d", &minor, &patch, &ex)
		if minor > version.minor{
			version = Version{
				3, minor, patch, ex,
			}
			path = string(v)
		}else if minor == version.minor{
			if patch > version.patch {
				version = Version{
					3, minor, patch, ex,
				}
				path = string(v)
			}else if patch == version.patch{
				if ex > version.ex{
					version = Version{
						3, minor, patch, ex,
					}
					path = string(v)
				}
			}
		}
	}
	path = strings.TrimSpace(path)
	fmt.Println("Find : ", version, path)
	return
}


type Version struct{
	major, minor, patch, ex int
}
func(s Version) String() string{
	return fmt.Sprintf("%d.%d.%d_%d", s.major, s.minor, s.patch, s.ex)
}
```
Unfortunatly, this code writen by MS notepad, and i can't use my laptop personal reason
I can't sure it works

I will test later

How do you think using go for env setting?

**Comment by deadprogram** [comment]
I prefer to stick with Makefiles and shell scripts, personally.

**Comment by deadprogram** [comment]
The compile issue within `imgproc.cpp` has been addressed in the `dev` branch. I will close it after the next release.

**Comment by deadprogram** [comment]
This code has now been released as part of v0.12.0 so now closing. Thanks again!
